lib/ansible/module_utils/basic.py

```diff
@@ -783,6 +784,9 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self._deprecations = []
         self._clean = {}
 
+        # Change to 'error' in Ansible 2.12
+        self._string_conversion_action = 'warn'
+
```

Comment:
you don't need to set default here, it already was set in configuration definition and should always be passed in

Reply:
If no default is set it could be problematic for debugging modules if the user doesn't set `_ansible_string_conversion_action` in their args.json file.

Reply:
not really, since the 'if' already treats 'any non ignore/error' as warn

Reply:
True, should have read that part first before commenting.